vowpalwabbit/reduction_stack.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "reductions_fwd.h"
+
+struct vw;
+
+typedef VW::LEARNER::base_learner* (*reduction_setup_fn)(VW::setup_base_fn&, VW::config::options_i&, vw&);
+
+namespace VW
+{
+struct default_reduction_stack_setup : public setup_base_fn
+{
+  std::vector<std::string> enabled_reductions;
```

Comment:
Public fields is usually bad.

Reply:
I'll address this with https://github.com/VowpalWabbit/vowpal_wabbit/pull/3176

Reply:
done